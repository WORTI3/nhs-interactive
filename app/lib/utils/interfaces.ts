import { SearchQuery } from "@/types";

export interface SearchProps {
  onSearch: (searchResult: SearchQuery) => void;
  onResultUpdate: (answer: string) => void;
  onDone: (done: boolean) => void;
}

export interface ResultProps {
  searchQuery: SearchQuery;
  answer: string;
  done: boolean;
  onReset: () => void;
}