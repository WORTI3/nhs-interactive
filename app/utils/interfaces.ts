import { SearchQuery } from "@/types";

export interface SearchProps {
  onSearch: (searchResult: SearchQuery) => void;
  onResultUpdate: (answer: string) => void;
  onDone: (done: boolean) => void;
  setResultIdStore: (resultIdStore: string) => void;
}

export interface ResultProps {
  searchQuery: SearchQuery;
  answer: string;
  done: boolean;
  onReset: () => void;
  answerIdStore: string;
}
