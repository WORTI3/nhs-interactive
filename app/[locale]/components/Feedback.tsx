import { Dialog, DialogTitle } from '@headlessui/react';
import { useState } from 'react';
import { Button } from './nhs';
import { useTranslations } from 'next-intl';

const FeedbackDialog = ({ isOpen, onClose, onSubmit }: {isOpen: any, onClose: any, onSubmit: any}) => {
  const translate = useTranslations('search');
  const [comment, setComment] = useState('');
  const [helpful, setHelpful] = useState(false);
  const [error, setError] = useState<string>("");

  const handleSubmit = () => {
    onSubmit({ comment, helpful });
    onClose();
  };

  return (
    <Dialog open={isOpen} onClose={onClose} className="fixed inset-0 overflow-hidden">
      <div className="absolute inset-0 overflow-hidden">
        <div className="absolute inset-0 bg-black opacity-50" />
        <div className="fixed inset-y-0 right-0 max-w-full flex">
          <div className="w-screen max-w-md">
            <div className="h-full bg-white p-4 shadow-xl">
              <DialogTitle>Provide Feedback</DialogTitle>
              <textarea
                className="w-full border rounded mt-4 p-2"
                placeholder="Leave a note..."
                value={comment}
                onChange={(e) => setComment(e.target.value)}
              />
              <div className="mt-4">
                <label>
                  <input
                    type="checkbox"
                    checked={helpful}
                    onChange={(e) => setHelpful(e.target.checked)}
                  />
                  Was this helpful?
                </label>
              </div>
              <div className="mt-4">
                <button
                  className="bg-blue-500 text-white p-2 rounded"
                  onClick={handleSubmit}
                >
                  Submit
                </button>
                <Button
                  id='submit'
                  as='a'
                  data-prevent-double-click
                  onClick={handleSubmit}
                  disabled={!!error && comment === ''}
                >
                  {translate('button.submit')}
                </Button>
                <button
                  className="ml-2 p-2"
                  onClick={onClose}
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Dialog>
  );
};

export default FeedbackDialog;
