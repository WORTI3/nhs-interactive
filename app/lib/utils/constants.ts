export const LOCAL_KEY_ID = "interactive-local-key";
export const UNRELATED_ANSWER = "not-answerable";
export const LOCAL_DB_URL = "http://127.0.0.1:8090";

export enum OpenAIModel {
  DAVINCI_TURBO = "gpt-3.5-turbo",
};