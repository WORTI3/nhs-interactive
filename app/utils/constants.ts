export const LOCAL_KEY_ID = "interactive-local-key";
export const UNRELATED_ANSWER = "not-answerable";
export const LOCAL_DB_URL = "http://127.0.0.1:8090";
export const SERVICE_NAME = "NHS Interactive";
export const SERVICE_DESC = "AI powered search.";
export const SERVICE_URL = "https://nhs-interactive.vercel.app";
export const LOCALES = ['en', 'cy', 'gd'];
export const USE_AI_RESPONSE_KEY = "use-ai-response";

export enum OpenAIModel {
  DAVINCI_TURBO = "gpt-3.5-turbo",
};