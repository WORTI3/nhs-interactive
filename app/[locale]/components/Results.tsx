import { FC } from "react";
import { UNRELATED_ANSWER } from "../../utils/constants";
import { ResultProps } from "../../utils/interfaces";
import { useApiLog } from "../../utils/hooks";
import { LogData } from "@/types";
import { useTranslations } from "next-intl";

export const Results: FC<ResultProps> = ({ searchQuery, answer, done, onReset }) => {
  const translate = useTranslations('results');
  const logData: LogData = { searchQuery, answer };
  useApiLog(logData, done);
  return (
    <>
      <p className='text-md text-nhs-blue mb-2'>{translate('question')}</p>
      <p className='italic'>{searchQuery.query} (?)</p>
      <hr className='nhsuk-section-break nhsuk-section-break--m nhsuk-section-break--visible' />
      { answer === UNRELATED_ANSWER ? <>
        <p>{translate('error.paragraphOne')}</p>
        <div className='nhsuk-details__text mb-6'>
          <p>{translate('error.paragraphTwo')}</p>
          <ul>
            <li>{translate('error.list.one')}</li>
            <li>{translate('error.list.two')}</li>
            <li>{translate('error.list.three')}</li>
          </ul>
        </div>
        </> : <>
          <p className='text-nhs-blue mb-2'>{translate('answer')}</p>
          <p className='overflow-auto'>
            {answer}
          </p>
        </>
      }
      
      {(done && answer !== UNRELATED_ANSWER) && (
        <>
          <hr className='nhsuk-section-break nhsuk-section-break--m nhsuk-section-break--visible' />
          <p className='text-nhs-blue mb-2'>{translate('sources')}</p>
          {searchQuery.sourceLinks.map((source, index) => (
            <div key={index} className='my-2 overflow-auto max-sm:flex'>
              <a
                className='hover:cursor-pointer hover:underline'
                target='_blank'
                rel='noopener noreferrer'
                href={source}
              >
                <span className="max-sm:flex-1  ">
                  {searchQuery.sourceHeadings[index] ?? 'NHS Search'}
                  <span className="ml-4 max-sm:float-none sm:float-right inline-flex items-center rounded-md bg-sky-50 max-sm:px-1.5 px-2 py-1 max-sm:text-xs text-xs max-sm:font-semibold font-bold text-sky-600 ring-1 ring-inset ring-sky-600/90 hover:bg-sky-100">
                    Verified source: {source.split("//")[1].split("/")[0].replace("www.", "")}
                  </span>
                </span>
              </a>
            </div>
          ))}
          <hr className='nhsuk-section-break nhsuk-section-break--m nhsuk-section-break--visible' />
        </>
      )}
      { (done || answer === UNRELATED_ANSWER) && (
        <button className="nhsuk-button" onClick={onReset}>{translate('button.newQuery')}</button>
      )}
    </>
  );
};
