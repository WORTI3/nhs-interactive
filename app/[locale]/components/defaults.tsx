import { OpenAIModel } from "../../utils/constants";

/**
 * `export const ModelInfo` is defining a React functional component that returns a paragraph element displaying
 * information about a specific OpenAI model. The model information is accessed through the `OpenAIModel` constant defined
 * in the `constants.ts` file. This component can be imported and used in other parts of the codebase.
 * 
 * @function
 * @name ModelInfo
 * @kind variable
 * @returns {JSX.Element}
 * @exports
 */
export const ModelInfo = () => {
  return (
    <p className="nhsuk-body-s nhsuk-u-secondary-text-color nhsuk-u-margin-top-7 nhsuk-u-margin-bottom-0">
      Model: { OpenAIModel.DAVINCI_TURBO }
    </p>
  );
}

/**
 * This code exports a function named `makeSourcesLinks` that takes in two parameters: `answer` and `sourceLinks`. The
 * function splits the `answer` string using a regular expression and maps over the resulting array. If a part of the
 * string matches the regular expression `/\[[0-9]+\]/`, it extracts the number within the square brackets, uses it to
 * retrieve a link from the `sourceLinks` array, and returns an anchor tag with the link and the original part of the
 * string. If the part of the string does not match the regular expression, it returns the original part of the string. The
 * function then returns an array of elements that can be rendered in the UI.
 * 
 * @function
 * @name makeSourcesLinks
 * @kind variable
 * @exports
 */
export const makeSourcesLinks = (answer: string, sourceLinks: string[]) => {
  const elements = answer.split(/(\[[0-9]+\])/).map((part, index) => {
    if (/\[[0-9]+\]/.test(part)) {
      const link = sourceLinks[parseInt(part.replace(/[\[\]]/g, "")) - 1];

      return (
        <a
          key={index}
          className="hover:cursor-pointer text-nhs-blue text-base"
          href={link}
          target="_blank"
          rel="noopener noreferrer"
        >
          {part}
        </a>
      );
    } else {
      return part;
    }
  });

  return elements;
};