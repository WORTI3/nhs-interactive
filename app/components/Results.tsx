import { Button } from "nhsuk-react-components";
import { FC } from "react";
import { makeSourcesLinks } from "./defaults";
import { UNRELATED_ANSWER } from "../lib/utils/constants";
import { ResultProps } from "../lib/utils/interfaces";
import { useApiLog } from "../lib/utils/hooks";
import { LogData } from "@/types";

export const Results: FC<ResultProps> = ({ searchQuery, answer, done, onReset }) => {
  const logData: LogData = { searchQuery, answer };
  useApiLog(logData, done);
  return (
    <>
      <p className='text-md text-nhs-blue'>Question</p>
      <p className='italic'>{searchQuery.query} (?)</p>
      <hr className='nhsuk-section-break nhsuk-section-break--m nhsuk-section-break--visible' />
      { answer === UNRELATED_ANSWER ? <>
        <p>It looks like your query may not be answerable.</p>
        <div className='nhsuk-details__text mb-6'>
          <p>This could be for several reasons:</p>
          <ul>
            <li>your query is not health related</li>
            <li>your query contained inappropriate language</li>
            <li>your query did not give enough detail</li>
          </ul>
        </div>
        </> : <>
          <p className='text-nhs-blue mb-2'>Answer</p>
          <p className='overflow-auto'>
            {makeSourcesLinks(answer, searchQuery.sourceLinks)}
          </p>
        </>
      }
      
      {done && (
        <>
          <hr className='nhsuk-section-break nhsuk-section-break--m nhsuk-section-break--visible' />
          <p className='text-nhs-blue'>Trusted Sources</p>
          {searchQuery.sourceLinks.map((source, index) => (
            <div key={index} className='mt-1 overflow-auto'>
              {`[${index + 1}] `}
              <a
                className='hover:cursor-pointer hover:underline'
                target='_blank'
                rel='noopener noreferrer'
                href={source}
              >
                {source.split("//")[1].split("/")[0].replace("www.", "")}
              </a>
            </div>
          ))}
          <hr className='nhsuk-section-break nhsuk-section-break--m nhsuk-section-break--visible' />
        </>
      )}
      { (done || answer === UNRELATED_ANSWER) && (
        <Button onClick={onReset}>Ask New Question</Button>
      )}
    </>
  );
};
