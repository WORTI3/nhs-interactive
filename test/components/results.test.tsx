import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Results } from '../../app/components/Results';
import { UNRELATED_ANSWER } from '../../app/lib/utils/constants';
import { ResultProps } from '@/app/lib/utils/interfaces';
import { load } from 'cheerio';

describe('Results', () => {
  let props: ResultProps;

  beforeEach(() => {
    props = {
      searchQuery: {
        query: 'test query',
        sourceLinks: ['https://www.link1', 'https://www.link2']
      },
      answer: 'test answer',
      done: true,
      onReset: jest.fn()
    };
  });
  
  test('rendered component should match snapshot', () => {
    const { container } = render(<Results {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('renders correctly with unrelated answer', () => {
    props.answer = UNRELATED_ANSWER;
    const { container } = render(<Results {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('renders correctly with correct elements', () => {
    const { container } = render(<Results {...props} />);
    const html = container.innerHTML;
    const $ = load(html);
  
    expect($('a').length).toBe(2); // 2 links
    expect($('p').length).toBe(5);
  });
});
