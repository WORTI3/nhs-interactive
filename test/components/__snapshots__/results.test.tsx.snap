// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results rendered component should match snapshot (cy) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Cwestiwn
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Newid
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            cwestiwn
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Ateb
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results rendered component should match snapshot (en) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Question
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Change
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            question
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Answer
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results rendered component should match snapshot (gd) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Ceist
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Atharraich
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            ceist
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Freagairt
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results renders correctly with unrelated answer (cy) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Cwestiwn
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Newid
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            cwestiwn
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p>
    mae'n ymddangos nad yw eich ymholiad yn atebadwy.
  </p>
  <div
    class="nhsuk-details__text mb-6"
  >
    <p>
      Gallai hyn fod am nifer o resymau:
    </p>
    <ul>
      <li>
        nid yw eich ymholiad yn ymwneud â iechyd
      </li>
      <li>
        roedd eich ymholiad yn cynnwys iaith anaddas
      </li>
      <li>
        nid oedd digon o fanylion yn eich ymholiad
      </li>
    </ul>
  </div>
  <a
    class="nhsuk-button   "
    href="/"
  >
    Gofyn cwestiwn newydd
  </a>
</div>
`;

exports[`Results renders correctly with unrelated answer (en) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Question
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Change
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            question
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p>
    It looks like your query may not be answerable.
  </p>
  <div
    class="nhsuk-details__text mb-6"
  >
    <p>
      This could be for several reasons:
    </p>
    <ul>
      <li>
        your query is not health related
      </li>
      <li>
        your query contained inappropriate language
      </li>
      <li>
        your query did not give enough detail
      </li>
    </ul>
  </div>
  <a
    class="nhsuk-button   "
    href="/"
  >
    Ask new question
  </a>
</div>
`;

exports[`Results renders correctly with unrelated answer (gd) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Ceist
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Atharraich
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            ceist
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p>
    tha coltas gum b'urrainn nach biodh freagairt agad air do chuirimh.
  </p>
  <div
    class="nhsuk-details__text mb-6"
  >
    <p>
      Dh'fhaodadh seo a bhith airson grunn fiosrachaidhean:
    </p>
    <ul>
      <li>
        chan eil do chuirimh a' buntainn ri slàinte
      </li>
      <li>
        tha cànan sgrìobhte nach bu chòir ann do chuirimh
      </li>
      <li>
        cha tugadh tu digriannan gu leòr le do chuirimh
      </li>
    </ul>
  </div>
  <a
    class="nhsuk-button   "
    href="/"
  >
    Cur ceist ùr
  </a>
</div>
`;

exports[`Results renders correctly with unrelated answer when source is also present (cy) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Cwestiwn
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Newid
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            cwestiwn
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p>
    mae'n ymddangos nad yw eich ymholiad yn atebadwy.
  </p>
  <div
    class="nhsuk-details__text mb-6"
  >
    <p>
      Gallai hyn fod am nifer o resymau:
    </p>
    <ul>
      <li>
        nid yw eich ymholiad yn ymwneud â iechyd
      </li>
      <li>
        roedd eich ymholiad yn cynnwys iaith anaddas
      </li>
      <li>
        nid oedd digon o fanylion yn eich ymholiad
      </li>
    </ul>
  </div>
  <a
    class="nhsuk-button   "
    href="/"
  >
    Gofyn cwestiwn newydd
  </a>
</div>
`;

exports[`Results renders correctly with unrelated answer when source is also present (cy) 2`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Cwestiwn
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Newid
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            cwestiwn
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Ateb
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results renders correctly with unrelated answer when source is also present (en) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Question
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Change
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            question
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p>
    It looks like your query may not be answerable.
  </p>
  <div
    class="nhsuk-details__text mb-6"
  >
    <p>
      This could be for several reasons:
    </p>
    <ul>
      <li>
        your query is not health related
      </li>
      <li>
        your query contained inappropriate language
      </li>
      <li>
        your query did not give enough detail
      </li>
    </ul>
  </div>
  <a
    class="nhsuk-button   "
    href="/"
  >
    Ask new question
  </a>
</div>
`;

exports[`Results renders correctly with unrelated answer when source is also present (en) 2`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Question
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Change
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            question
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Answer
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results renders correctly with unrelated answer when source is also present (gd) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Ceist
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Atharraich
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            ceist
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p>
    tha coltas gum b'urrainn nach biodh freagairt agad air do chuirimh.
  </p>
  <div
    class="nhsuk-details__text mb-6"
  >
    <p>
      Dh'fhaodadh seo a bhith airson grunn fiosrachaidhean:
    </p>
    <ul>
      <li>
        chan eil do chuirimh a' buntainn ri slàinte
      </li>
      <li>
        tha cànan sgrìobhte nach bu chòir ann do chuirimh
      </li>
      <li>
        cha tugadh tu digriannan gu leòr le do chuirimh
      </li>
    </ul>
  </div>
  <a
    class="nhsuk-button   "
    href="/"
  >
    Cur ceist ùr
  </a>
</div>
`;

exports[`Results renders correctly with unrelated answer when source is also present (gd) 2`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Ceist
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Atharraich
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            ceist
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Freagairt
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results renders correctly with valid answer (cy) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Cwestiwn
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Newid
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            cwestiwn
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Ateb
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results renders correctly with valid answer (en) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Question
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Change
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            question
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Answer
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;

exports[`Results renders correctly with valid answer (gd) 1`] = `
<div>
  <p
    class="text-md text-nhs-blue mb-2"
  >
    Ceist
  </p>
  <dl
    class="nhsuk-summary-list"
  >
    <div
      class="nhsuk-summary-list__row"
    >
      <dd
        class="nhsuk-summary-list__value"
      >
        test query
         (?)
      </dd>
      <dd
        class="nhsuk-summary-list__actions"
      >
        <a
          href="/?q=test+query#search-input"
        >
          Atharraich
          <span
            class="nhsuk-u-visually-hidden"
          >
             
            ceist
          </span>
        </a>
      </dd>
    </div>
  </dl>
  <p
    class="text-nhs-blue mb-2"
  >
    Freagairt
  </p>
  <p
    class="overflow-auto"
  >
    test answer
  </p>
</div>
`;
